### ISSUE TITLE
How high BLEU scores will be achieved provided enough training iterations ?

### ISSUE THREAD

**Issue by MenSanYan** [DeepRNN/image_captioning#11]
Can your code achieve the same or similar BLEU scores published in the original paper?

**Comment by loganlebanoff** [comment]
The highest BLEU scores I got were:

Bleu_1: 0.607
Bleu_2: 0.416
Bleu_3: 0.275
Bleu_4: 0.181

This was with a pretrained vgg-16 model and pretrained word embeddings. I don't know what settings to use to get the same results as the original paper.

**Comment by MenSanYan** [comment]
@loganlebanoff Thanks. However, the bleu scores are far from the results in the original paper. It seems more difficult to tune hyperparameters than to design an effective model. @DeepRNN Can you provide some instructions to achieve state-of-the-art results ? 怎么才能获得和原论文类似的结果呢？如何才能使用您的代码取得最好的结果呢？求帮助。

**Comment by aayushARM** [comment]
@loganlebanoff Can you understand what exactly is this line doing in def attend() function in models.py?:

` masked_alpha = alpha * tf.tile(tf.expand_dims(masks[:, idx], 1), [1, self.num_ctx])`

**Comment by aayushARM** [comment]
@loganlebanoff @DeepRNN 
I cannot seem to understand this line from build_rnn() in model.py(line no. 266):

tiled_masks = tf.tile(tf.expand_dims(masks[:, idx], 1), [1, self.num_ctx]) 

can anyone explain what are we doing here and why do we need to do this? Any help will be appreciated!

**Comment by MikhailovSergei** [comment]
Hello, after how much iteration u achieve this result?

**Comment by Kinghup** [comment]
> The highest BLEU scores I got were:
> 
> Bleu_1: 0.607
> Bleu_2: 0.416
> Bleu_3: 0.275
> Bleu_4: 0.181
> 
> This was with a pretrained vgg-16 model and pretrained word embeddings. I don't know what settings to use to get the same results as the original paper.

I eval my model but the results is strange. i don't know where is wrong. do you have any idea?

![image](https://user-images.githubusercontent.com/38780568/53380180-76fb1980-39a7-11e9-86e4-7762b98866b6.png)

![image](https://user-images.githubusercontent.com/38780568/53380181-7793b000-39a7-11e9-98b3-562a0da3d9e8.png)
